lms/templates/courseware/progress.html

```diff
@@ -169,20 +171,33 @@ <h3 class="hd hd-3" id="chapter_${loop.index}">${ chapter['display_name']}</h3>
                                 %for section in chapter['sections']:
                                     <div>
                                         <%
+                                        hide_url_date = (section.self_paced and section.end) or section.due
+                                        hide_url = not staff_access and section.hide_after_due and hide_url_date and datetime.now(UTC) > hide_url_date
```

Comment:
i noticed there is some duplication here .. perhaps not a big deal, though you could potentially outsource the code into a shared function and then pass it into the template context

Reply:
Yeah, I don't think that's worth it as this bit of code is purely to support the legacy view which we are hoping to move away from soon anyway. And then it might be weird that our serializer is calling out to some utility function that only it uses.